faunadb-scala/src/main/scala/faunadb/values/Value.scala

```diff
@@ -124,9 +123,74 @@ object BooleanV {
 // Fauna special types
 
 /** A Ref. */
-case class RefV(@(JsonProperty @field @param)("@ref") value: String) extends ScalarValue
+case class RefV(@(JsonIgnore @getter) id: String,
+                @(JsonIgnore @getter) clazz: Option[RefV] = None,
+                @(JsonIgnore @getter) database: Option[RefV] = None) extends ScalarValue {
+
+  @JsonProperty("@ref")
+  lazy val refValue: Any = RefID(id, clazz, database)
+
+  @JsonInclude(JsonInclude.Include.NON_ABSENT)
+  private case class RefID(
+     @(JsonProperty @getter)("id") id: String,
+     @(JsonProperty @getter)("class") clazz: Option[RefV],
+     @(JsonProperty @getter)("database") database: Option[RefV])
+}
+
 object RefV {
-  def apply(clss: RefV, id: String): RefV = RefV(s"${clss.value}/$id")
+  def apply(id: String, clazz: RefV): RefV = RefV(id, Option(clazz))
+}
+
+/** A Database reference */
+object DatabaseV {
```

Comment:
Same question here as in java driver: is there any significant advantage in having this value constructors vs enforce the usage of the expression level constructors?

Reply:
removing those constructors, specially in Scala will make it really hard to create references:
```
RefV("123", ClassV("a_class", DatabaseV("a_database")))
```
will turn to:
```
RefV("123", Some(RefV("a_class", Native.Classes, Some(RefV("a_database", Native.Databases)))))
```

Reply:
I feel like those constructors are most useful for us while testing the driver. We don't have static references in our docs, for example. We usually tell people to create their references at the expression level by calling `Ref(class, id)`.

Reply:
We could add constructors to the `RefV` in order to make our lives easier. But in general, the driver should point people towards expressions more than towards values.